RELEASE/scripts/sl_ascend/sl_tcrs.ash

```diff
@@ -203,10 +208,17 @@ boolean sl_knapsackAutoEat(boolean simulate)
 				int n = count(fullness);
 				fullness[n] = it.fullness;
 				adv[n] = expectedAdventuresFrom(it);
+				adv[n] += min(1.0, item_amount($item[special seasoning]) / limit);
+				if (item_amount(it) == 0 && creatable_amount(it) > 0)
+				{
+				  int turns_to_craft = creatable_turns(it, 1, false);

```

Comment:
The reason I said to do `creatable_turns(it, crafted, false) - creatable_turns(it, crafted - 1, false);` instead of just `creatable_turns(it, 1, false);` is that you might theoretically (especially with van keys or duffel bags in the equation) be able to craft 3 of something that takes two turns to craft, but already have 1 of it with the first stage of crafting done. It's not exactly a huge difference, but might as well be as precise as possible.

Also, `item_amount(it) == 0)` needs to be `item_amount(it) < i + 1`, since we might also already have one fully crafted or something like that.

Then the `crafted` in the above code would be `i + 1 - item_amount(it)`.

Reply:
The latest revision distinguishes between "I want to craft these" and "I already have these" as part of prioritizing already-owned items. I think the code implements what you're imagining.